core-bundle/src/EventListener/SubrequestCacheListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Contao\CoreBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\HttpCache\ResponseCacheStrategy;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+class SubrequestCacheListener
+{
+    public const MERGE_CACHE_HEADER = 'Sf-Merge-Cache-Control';
+
+    /**
+     * @var ResponseCacheStrategy[]
+     */
+    private $strategyStack = [];
```

Comment:
Why do we need a stack here? As far as I understood from our latest discussions there's only one master request in the same process unless you're using `HttpCache` in which case the services are reset for every master request. So imho, it should implement the `ResetInterface`.

Reply:
theoretically, we should not need a stack, as every master request should stack and pop the strategy. But I'm not sure if there could a reason this is bypassed (e.g. an exception?), so a stack will make sure even a stacking of main requests will work.

Reply:
Okay, but we should still reset the stack, no?

Reply:
we do? https://github.com/contao/contao/pull/718/files#diff-c53e2a2fc44f16c19b46cb6d977e7ccfR73

Reply:
I meant service resetting :)

Reply:
implemented in c0ce1fd653459b850b7553e893061372870dc093